operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/impl/AbstractSystemConfigurationImpl.java

```diff
@@ -0,0 +1,283 @@
+
+    // The following are set on startup:
+    private static final int coreCosMaxResources = defaultCoreCosMaxResources();
+    private static final int coreCosMinSize = defaultCoreCosMinSize();
+    private static final int coreCosMaxSize = defaultCoreCosMaxSize();
+    private static final boolean coreCosUseServerTruststore = defaultCoreCosUseServerTruststore();
+    private static final int coreCosRequestTimeout = defaultCoreCosRequestTimeout();
+    private static final int coreCosSocketTimeout = defaultCoreCosSocketTimeout();
+    private static final String coreBatchIdEncryptionKey = defaultCoreBatchIdEncryptionKey();
+    private static final int coreMaxParititions = defaultCoreMaxParititions();
+    private static final int inputLimits = defaultInputLimits();
+
+    @Override
+    public boolean getCoreCosTcpKeepAlive() {
+        return true;
+    }
+
+    @Override
+    public int getCoreMaxPartitions() {
+        return coreMaxParititions;
+    }
+
+    private static final int defaultCoreMaxParititions() {
+        return FHIRConfigHelper.getIntProperty("fhirServer/bulkdata/core/max-partitions", MAX_PARTITIONPROCESSING_THREADNUMBER);
+    }
+
+    @Override
+    public String getCoreBatchIdEncryptionKey() {
+        return coreBatchIdEncryptionKey;
+    }
+
+    private static final String defaultCoreBatchIdEncryptionKey() {
+        return FHIRConfigHelper.getStringProperty("fhirServer/bulkdata/core/batch-id-encryption-key", null);
+    }
+
+    @Override
+    public int getCoreCosSocketTimeout() {
+        return coreCosSocketTimeout;
+    }
+
+    private static final int defaultCoreCosSocketTimeout() {
+        return FHIRConfigHelper.getIntProperty("fhirServer/bulkdata/core/cos/socket-timeout", 12000);
+    }
+
+    @Override
+    public int getInputLimit() {
+        return inputLimits;
+    }
+
+    private static final int defaultInputLimits() {
+        return FHIRConfigHelper.getIntProperty("fhirServer/bulkdata/core/max-inputs", OperationConstants.IMPORT_MAX_DEFAULT_INPUTS);
+    }
+
+    @Override
+    public int getCoreCosRequestTimeout() {
+        return coreCosRequestTimeout;
+    }
+
+    private static final int defaultCoreCosRequestTimeout() {
+        return FHIRConfigHelper.getIntProperty("fhirServer/bulkdata/core/cos/request-timeout", 10000);
+    }
+
+    @Override
+    public boolean shouldCoreCosUseServerTruststore() {
+        return coreCosUseServerTruststore;
+    }
+
+    private static final boolean defaultCoreCosUseServerTruststore() {
+        return FHIRConfigHelper.getBooleanProperty("fhirServer/bulkdata/core/cos/use-server-truststore", Boolean.TRUE);
+    }
+
+    @Override
+    public int getCoreCosMaxSize() {
+        return coreCosMaxSize;
+    }
+
+    private static final int defaultCoreCosMaxSize() {
+        final String PATH = "fhirServer/bulkdata/core/cos/max-size";
+        return FHIRConfigHelper.getIntProperty(PATH, DEFAULT_COSFILE_MAX_SIZE);
+    }
```

Comment:
I'm thinking that we should probably make this a `long`.
The default fits safely into an int, but--if we keep this in bytes--then its not too far off from Integer.MAX_VALUE:
`209715200`  (our default; 200MiB)
`2147483647` (Integer.MAX_VALUE; 2048 MiB)

Unfortunately, our fhir-config stuff doesn't currently support Long properties, so we'd either need to add that support or decide to represent the value in KB or MB (or MiB) in our config.

Reply:
I think my vote would be to just specify it in MiB in our config.  I don't think a max-size of <1 MiB makes any sense, and I don't think there's much value in allowing more granular values.

Reply:
I've changed to MIB in config

Reply:
It's in the next commit